libr/core/cbin.c

```diff
@@ -2491,6 +2491,46 @@ struct io_bin_section_info_t {
 	int fd;
 };
 
+/* Map Sections to Segments https://github.com/radareorg/radare2/issues/14647 */
+static int bin_map_sections_to_segments(RBin *bin) {
+	RListIter *iter, *iter2;
+	RBinSection *section = NULL, *segment = NULL;
+	RList *sections = r_list_new();
+	RList *segments = r_list_new();
+	RList *tmp = r_bin_get_sections (bin);
+
+	printf ("Section to Segment mapping:\n");
+	RTable *table = r_table_new();
+	RTableColumnType *typeString = r_table_type ("string");
+
+	r_table_add_column (table, typeString, "Segment", 0);
+	r_table_add_column (table, typeString, "Section", 0);
+
+	r_list_foreach(tmp, iter, section) {
+		if (section->is_segment) {
```

Comment:
Use the ternary operator:

RList *list = section->is_segment? segments: sections;
r_list_append(list, section);